### PR TITLE
Rename binaries to moby-*

### PR DESCRIPTION
**Pull request by tiborvass**
Since the project and product separation between moby and docker, we had yet to rename the dockerd binary to a moby-affiliated name. It is in accordance with the plan in https://github.com/moby/moby/issues/35115

**This PR does not impact [docker CE](https://github.com/docker/docker-ce).**

The complete list of binaries being renamed in this repository:
```
dockerd			-> moby-engine
docker-init		-> moby-tini
docker-proxy		-> moby-proxy
docker-runc		-> moby-runc
docker-containerd	-> moby-containerd
docker-containerd-ctr	-> moby-containerd-ctr
docker-containerd-shim	-> moby-containerd-shim
```

Again, this should not affect Docker users.

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
I'd prefer daemon over server :smiling_imp:

File: cmd/server/docker.go
Comment (AkihiroSuda) [inline cmd/server/docker.go]: Docker -> Moby Core? (should be definable on compilation-time)

**Comment by AkihiroSuda** [comment]
cmd/server/README.md seems unneeded

**Comment by AkihiroSuda** [comment]
What's the current status?

**Comment by justincormack** [comment]
Needs a rebase

**Comment by AkihiroSuda** [comment]
ping @tiborvass @mavenugo 
What's current status?

(I think I have seen concrete schedule on this somewhere else, but could not find that again .. :sweat_smile: )

**Comment by tiborvass** [comment]
@AkihiroSuda feel free to push to this branch, I wanted to get back to this but need to finish more pressing things first.

**Comment by justincormack** [comment]
Needs a rebase again. Lets prioritise this.

**Comment by thaJeztah** [comment]
After switching to containerd 1.0, will we drop the prefixed names?

**Comment by justincormack** [comment]
@thaJeztah I think we expect people to start packaging moby with upstream `runc` and `containerd` so potentially yes.

**Comment by AkihiroSuda** [comment]
`moby-init` might be confusing. Can we use `moby-engine-%s` prefix?

**Comment by justincormack** [comment]
@AkihiroSuda we are now using unmodified upstream `tini` as well, I think we could just use that name.

**Comment by AkihiroSuda** [comment]
Not sure for `tini`, but for `runc` and `containerd`, using them without prefix might cause issues..

Can we put these binaries to `/usr/local/libexec/moby-engine` (Or `/usr/local/libexec/moby`)?

e.g.
`/usr/local/libexec/moby-engine/runc`, `/usr/local/libexec/moby-engine/containerd`,  and `/usr/local/libexec/moby-engine/tini`

**Comment by justincormack** [comment]
@AkihiroSuda we should open another issue to discuss those changes.

**Comment by AkihiroSuda** [comment]
Post a comment to https://github.com/containerd/containerd/issues/1508#issuecomment-335444885
(No need to use prefix nor libexec if we can support installing different versions of containerd simultaneously)

**Comment by cpuguy83** [comment]
```
15:43:27 daemon/runtime_autogen.go:7: DefaultRuntimeBinary redeclared in this block
15:43:27 	previous declaration at daemon/daemon_unix.go:61
```

Also a bunch of warnings from gosimple.

**Comment by dnephin** [comment]
I'm working on fixing the windowsRS1 entrypoint, which should let us fix that build.

The janky failure was a timeout, probably because of load on jenkins right now.  I'll try rebuilding if the other builds are green.

**Comment by dnephin** [comment]
windowsRS1 fix depends on #35362

**Comment by dnephin** [comment]
Build is green!

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]
lgtm

cc @tianon

**Review by thaJeztah — APPROVED** [review]
LGTM, but needs a rebase due to https://github.com/moby/moby/pull/35389 (sorry! 😅)

**Comment by dnephin** [comment]
rebased, and green again

**Comment by AkihiroSuda** [comment]
Can we have a blog article about this for announcement?

**Comment by thaJeztah** [comment]
Yes, a blog would be appropriate :smile: 

I would like to have some more eyes on this, before we merge, and see if it needs additional coordination.

** do not merge yet **

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

**Comment by AkihiroSuda** [comment]
What should we do with this?
https://github.com/moby/moby/blob/6e7071ba2ae1abc04fe3fa1b33713cf28981b341/contrib/selinux-fedora-24/docker-engine-selinux/docker.fc
https://github.com/moby/moby/blob/6e7071ba2ae1abc04fe3fa1b33713cf28981b341/contrib/selinux-oraclelinux-7/docker-engine-selinux/docker.fc

@justincormack @thaJeztah @andrewhsu

**Comment by AkihiroSuda** [comment]
@justincormack 
Should we remove or update bundle installers for "Docker for Mac" and Docker Machine?

https://github.com/moby/moby/blob/6e7071ba2ae1abc04fe3fa1b33713cf28981b341/contrib/mac-install-bundle.sh
https://github.com/moby/moby/blob/6e7071ba2ae1abc04fe3fa1b33713cf28981b341/contrib/docker-machine-install-bundle.sh

(Either way, these are just contrib scripts and no need to be changed in this PR 😄 )

**Comment by thaJeztah** [comment]
@dnephin @tiborvass what are we gonna do for https://master.dockerproject.org ? I think the current setup is configured to;

- build the daemon from https://github.com/moby/moby master
- build the CLI from https://github.com/docker/cli master

Historically, that site built master builds for the Docker project, and I know it's used for internal e2e tests to catch regressions early (combination docker cli <--> moby master), but I think it's also referred to by Microsoft for people that want to try a "cutting edge" version of "Docker" on Windows.

The last one is a bit of a pain, because it's not "Docker".

/cc @seemethere

**Comment by dnephin** [comment]
We should probably have an nightly job that updates docker-ce and publishes from there.

**Comment by lsm5** [comment]
Any update on this? Also, are we at a point yet where we could use the distro's runc and containerd instead of docker-[runc,containerd] ? /cc @carlwgeorge

**Comment by AkihiroSuda** [comment]
Is this still on plan?

File: hack/make.ps1
Comment (jhowardmsft) [inline hack/make.ps1]: Why is this being removed?

**Comment by jhowardmsft** [comment]
This probably needs a change to the .rc files under `hack\make\.resources-windows` which embed the product name as seen in explorer. eg


image

**Comment by lsm5** [comment]
Is the rename still being considered?

**Comment by crosbymichael** [comment]
I'd say we close this as it's probably not worth it at this point.

**Pull request closed without merging**
